truss/templates/trtllm/packages/build_engine_utils.py

```diff
@@ -0,0 +1,35 @@
+from pathlib import Path
+
+from schema import EngineBuildArgs
+
+
+def build_engine_from_config_args(
+    engine_build_args: EngineBuildArgs,
+    dst: Path,
+):
+    import sys
+
+    sys.path.append("/app/baseten")
+
+    import os
+    import shutil
+
+    # NOTE: These are provided by the underlying base image
+    from build_engine import Engine, build_engine
+    from trtllm_utils import docker_tag_aware_file_cache
+
+    engine = Engine(**engine_build_args.model_dump())
```

Comment:
If engine is a pydantic model, it's more idiomatic to use the parse functions (V1 `parse_obj` V2 `model_validate`).

But I'm confused, truss repo uses `pydantic = "^1.10.9"` and `model_dump` is from the V2 API.

https://github.com/basetenlabs/truss/blob/main/pyproject.toml#L35C1-L35C9

Reply:
I think this is another artifact of the base image used. The base image has Pydantic installed. Also, that pyproject requirements are not used at run-time. What's used at run-time is this file:
https://github.com/basetenlabs/truss/blob/main/truss/templates/server/requirements.txt